### ISSUE TITLE
Clippy error

### ISSUE THREAD

**Issue by dalance** [jsinger67/parol#47]
I got the following clippy error in the generated code.

```
warning: this function has too many arguments (9/7)
    --> crates/parser/src/generated/veryl_grammar_trait.rs:5996:5
     |
5996 | /     fn if_statement(
5997 | |         &mut self,
5998 | |         _if: &ParseTreeStackEntry<'t>,
5999 | |         _expression: &ParseTreeStackEntry<'t>,
...    |
6005 | |         _parse_tree: &Tree<ParseTreeType<'t>>,
6006 | |     ) -> Result<()> {
     | |___________________^
     |
     = note: `#[warn(clippy::too_many_arguments)]` on by default
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments
```

Could you add `#![allow(clippy::too_many_arguments)]` like 4560fd21094217071fc78e640c42d5d8194c9a9c?

I think `#![allow(clippy::all)]` may be better because user can't fix clippy error in the generated code in any case.

**Comment by jsinger67** [comment]
Hi @dalance,
I'm aware of this problem and have already disabled certain clippy warnings in generated code.
The too_many_arguments warning is a warning that the user indeed could circumvent by reformulating his grammar so that productions have less elements on their right-hand side.
But I'm not sure if parol should impose this Rust related constraint onto its grammar definition language.
Therefore I think I will add a new CLI option and a corresponding builder configuration that will insert this directive into the generated code.
All other clippy warnings should be regarded as bugs in the code generation and fixed elsewhere.

**Comment by jsinger67** [comment]
I'm starting with the implementation. 😎

**Comment by jsinger67** [comment]
Fixed with commit [05c2a16](https://github.com/jsinger67/parol/commit/05c2a16b89d7ad998d7c23ba9f17b167b12cfb10)
Have fun 🎉
